### PR TITLE
Update Travis to use stages

### PR DESCRIPTION
**Pull request by dgarros**
This PR is mainly about updating travis to run in stages instead of running everything at once.
I also fixed some error reported by pylint, not sure why it didn't catch them previously and i added an additional test to ensure that the integration tests are passing.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,59 @@
...
-  - "invoke integration-tests"
+  - "invoke pytest"
+
+jobs:
+  include:
+    - stage: "lint"
+      env:
+        - "INVOKE_LOCAL=True"
+      before_script:
+        - "pip install invoke poetry toml"
```
Comment (nniehoff) [inline .travis.yml:33]: are you sure you want to pip install poetry?

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,59 @@
...
+
 language: "python"
-services:
-  - "docker"
 python:
   - 3.7
   - 3.8
 
-before_script:
-  - "pip install invoke toml poetry ansible pynetbox"
```
Comment (nniehoff) [inline .travis.yml:10]: Maybe I didn't follow far enough but why do we need ansible?

Comment (nniehoff) [reply .travis.yml:10]: pip install poetry?

File: .travis.yml (hunk 1 above)
Comment (nniehoff) [reply .travis.yml:33]: I thought there was recommendations against this, but it looks like it is valid.

File: network_importer/cli.py [hunk 3]
```diff
@@ -43,7 +43,6 @@
 @click.group()
 def main():
     """Main CLI command for the network_importer."""
-    pass
```
Comment (jvanderaa) [inline network_importer/cli.py:46]: Was this part of the change as well?

**Review by jvanderaa — COMMENTED** [review]
LG outside of what is commented already

File: .travis.yml (hunk 1 above)
Comment (dgarros) [reply .travis.yml:33]: As far as I understand it's ok to do that in docker and in closed environment like travis

File: .travis.yml (hunk 2 above)
Comment (dgarros) [reply .travis.yml:10]: we are using ansible during the integration tests to populate the initial inventory in Netbox

File: network_importer/cli.py (hunk 3 above)
Comment (dgarros) [reply network_importer/cli.py:46]: Not sure why pylint didn't catch this one earlier but I think it's correct that we don't need pass if we have the docstring

**Review by nniehoff — APPROVED** [review]
LGTM

**Pull request merged**